pkg/internal/enhancedmetrics/registry.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2026 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package enhancedmetrics
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/prometheus-community/yet-another-cloudwatch-exporter/pkg/internal/enhancedmetrics/service"
+	"github.com/prometheus-community/yet-another-cloudwatch-exporter/pkg/internal/enhancedmetrics/service/dynamodb"
+	"github.com/prometheus-community/yet-another-cloudwatch-exporter/pkg/internal/enhancedmetrics/service/elasticache"
+	"github.com/prometheus-community/yet-another-cloudwatch-exporter/pkg/internal/enhancedmetrics/service/lambda"
+	"github.com/prometheus-community/yet-another-cloudwatch-exporter/pkg/internal/enhancedmetrics/service/rds"
+)
+
+// DefaultRegistry is the default registry containing all built-in enhanced metrics services
+// It allows registering additional services if needed, or replacing existing ones, e.g. for testing purposes.
+//
+// Note:In the future, it can be removed in favor of being injected via dependency injection.
+// However, it will require changes in the YACE's API.
+var DefaultRegistry = (&Registry{}).
```

Comment:
I feel like the naming overlap with Prometheus registries is going to be confusing, maybe we should call it EnhancedMetricServiceRegistry to be more clear?